src/docs/index.jsx

```diff
@@ -561,51 +550,57 @@ const App = createClass({
 
 	searchResults() {
 		const { search } = this.state;
-
-		return _.flatMap(docgenMap, (value, componentName) => {
-			if (!this.showPrivateComponents() && value.isPrivateComponent) {
-				return [];
-			}
-
-			// exclude child components from search
-			if (_.includes(componentName, '.')) {
-				return [];
-			}
-
-			if (componentName.toLowerCase().indexOf(search.toLowerCase()) !== -1) {
-				return componentName;
+		const searchLower = _.toLower(search);
+		const showPrivateComponents = this.showPrivateComponents();
+
+		return _.filter(_.keys(lucidComponents), key => {
+			const componentRef = lucidComponents[key];
+			const isPrivate =
+				_.get(componentRef, 'peek.isPrivate') ||
+				_.get(componentRef, '_isPrivate');
+			if (showPrivateComponents || !isPrivate) {
+				return _.includes(_.toLower(key), searchLower);
 			}
-
-			return [];
 		});
 	},
 
 	render() {
 		const { search } = this.state;
 
-		const docgenGroups = _.reduce(
-			docgenMap,
-			(acc, value, key) => {
-				if (!this.showPrivateComponents() && value.isPrivateComponent) {
-					return acc;
-				}
+		const suggestions = this.searchResults();
 
-				// exclude child components from grouping
-				if (_.includes(key, '.')) {
-					return acc;
+		const categoryTree = _.reduce(
+			lucidComponents,
+			(acc, value, key) => {
+				const categories = _.get(value, 'peek.categories', []);
+				const path = categories.concat(key).join('.');
+				const isPrivate =
+					_.get(value, 'peek.isPrivate') || _.get(value, '_isPrivate');
+				if (this.showPrivateComponents() || !isPrivate) {
+					return _.set(acc, path, value);
 				}
-
-				const path = value.customData.categories.join('.');
-				const newGroup = _.get(acc, path, []);
-				newGroup.push(key);
-				return _.set(acc, path, newGroup);
+				return acc;
 			},
 			{}
 		);
-		const suggestions = this.searchResults();
 
 		return (
 			<div className="App">
+				{/* `.App-loading` helps prevent a FOUC */}
+				<div
+					className="App-loading"
+					style={{
+						position: 'fixed',
+						left: 0,
+						top: 0,
+						width: '100%',
+						height: '100%',
+						backgroundColor: 'white',
+						zIndex: 2,
+						pointerEvents: 'none',
+						transition: 'opacity 200ms ease 200ms',
```

Comment:
Neat!

Reply:
Any reason not to just put this in index.less since we're already styling it there too?

Reply:
The reason the FOUC happens in the first place is because the page renders before the styles are loaded.